### PR TITLE
Add automatic snapshot unlocking upon reboot

### PR DESCRIPTION
**Pull request by szaimen**
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: nextcloud_update.sh [hunk 1]
```diff
@@ -998,5 +1001,23 @@ Maintenance mode is kept on."
...
+            mapfile -t PROCESS_IDS_NEW <<< "$PROCESS_IDS_NEW"
+            for process in "${PROCESS_IDS_NEW[@]}"
+            do
+                print_text_in_color "$ICyan" "Killing the process with PID $process to prevent a potential automatic restart..."
+                if ! kill "$process"
+                then
+                    print_text_in_color "$IRed" "Couldn't kill the process with PID $process..."
+                fi
+            done
+        fi
```
Comment (szaimen) [inline nextcloud_update.sh:1020]: I've copied this from here: https://github.com/nextcloud/vm/blob/caa3b163b8ccadf782d7b055193094b31fb5d2ae/nextcloud_update.sh#L41-L55

**Comment by szaimen** [comment]
This is ready 👍

**Comment by enoch85** [comment]
It would be nice if you would explain exactly what it does. Sure I can read code, but it's easier if you explain. :)

**Comment by szaimen** [comment]
As you might know is the snapshot with the name NcVM-snapshot-pending currently preventing updates and backups & co to get executed if one of those already runs. It is kind of a blocker that prevents bad things from happening. If you shut down the VM while an update is running or while a backup is running, currently the next backup or update will be blocked from executing because the snapshot-pending still exists. To prevent those issues and make it easier to unlock the backup and update, I've added a cronjob that automatically unlocks backups and updates upon reboot.

**Comment by enoch85** [comment]
@szaimen Urgent: https://github.com/nextcloud/vm/runs/2403856054

**Comment by szaimen** [comment]
@enoch85 where does this error suddenly come from? It was all green when I've vreated the PR...

**Pull request merged**